site/src/pages/WorkspacePage/WorkspaceReadyPage.tsx

```diff
@@ -142,6 +125,70 @@ export const WorkspaceReadyPage = ({
     onError: onDeadlineChangeFails,
   });
 
+  // Auto start
+  const canAutostartResponse = useQuery(
+    workspaceResolveAutostart(workspace.id),
+  );
+  const canAutostart = !canAutostartResponse.data?.parameter_mismatch ?? false;
+
+  // SSH Prefix
+  const sshPrefixQuery = useQuery(deploymentSSHConfig());
+
+  // Favicon
+  const favicon = getFaviconByStatus(workspace.latest_build);
+  const [faviconTheme, setFaviconTheme] = useState<"light" | "dark">("dark");
+  useEffect(() => {
+    if (typeof window === "undefined" || !window.matchMedia) {
+      return;
+    }
+
+    const isDark = window.matchMedia("(prefers-color-scheme: dark)");
+    // We want the favicon the opposite of the theme.
+    setFaviconTheme(isDark.matches ? "light" : "dark");
+  }, []);
```

Comment:
I know that we don't have themes yet, but just for future-proofing, could this effect be updated to add an event listener to the `matchMedia` result? That way, it can change the favicon theme whenever the matcher detects a change?

Reply:
Yeap, but I think this is outside of the scope of this refactoring - I just moved this code.